### PR TITLE
Ask notif permission

### LINKED ISSUE #723: Notifications Blocked on Android 13
### Description

I noticed on Android 13 that notifications seem to be blocked by default. This is impacting our notifications feature, so even if you have notifications enabled for a podcast, you won't receive them unless you manually unblock notifications via Settings > Apps > Pocket Casts > Notifications. 

I wonder if this is an Android 13 bug or something we can force enable when notifications are switched on within the app

### Step-by-step reproduction instructions

1. Install Pocket Casts on a device running Android 13
2. Go to 'Settings > Apps > Pocket Casts > Notifications' and check what it is set to

### Screenshots or screen recording

Pixel 6 using Android 12, and Pixel 7 using Android 13: 

image image

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

Google Pixel 7, Android 13, Pocket Casts v7.30-rc-3 + Samsung S21 FE, Android 13, Pocket Casts v7.30-rc-3

### PR DESCRIPTION
**Pull request by xenom34**
## Description
This pull request is a implementation proposal regarding issue #723. The authorization check for notifications will be done upon starting the MainActivity (after the login and register interface), and will only run on devices equipped with Android 13 and higher.
Hope this works for you!😊

CC: @lucile98-ode (thanks for the help)

Fixes #723 

## Testing Instructions
1. Open the app
2. Register or login (or skip this step, it doesn't matter)
3. The app asks permission for sending notifications (won't show on your screen if you already launched with Android 13 and already approved this permission)

## Screenshots or Screencast 
image

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

File: app/src/main/AndroidManifest.xml
Comment (ashiagr) [inline app/src/main/AndroidManifest.xml:15]: This permission already exists in the manifest and can be removed.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (ashiagr) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:248]: Trying to show a snackbar before the content is set crashes the application. 

```
Caused by: java.lang.IllegalArgumentException: No suitable parent found from the given view. Please provide a valid view.
                            	at com.google.android.material.snackbar.Snackbar.makeInternal(Snackbar.java:198)
                            	at com.google.android.material.snackbar.Snackbar.make(Snackbar.java:157)
                            	at [au.com](http://au.com/).shiftyjelly.pocketcasts.ui.MainActivity.onCreate(MainActivity.kt:247)
```

Let us move the surrounding if-block below `setContentView(view)` and extract it into a separate method.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (ashiagr) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:250]: We might want to increase the snackbar duration (~5000ms) as there is an action set on it, it felt that snackbar dismissed a bit quickly before anyone can tap on the action.

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (ashiagr) [inline app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt:266]: Let's clean up some of the log messages here and at other places.

**Review by ashiagr — COMMENTED** [review]
Thank you for this PR, @xenom34! We really appreciate your contribution 🙏

I left a couple of comments to make it ready for merge. Also, you can run .`/gradlew :spotlessApply` on the root folder to fix formatting issues and make the CI happy.

I know it's just an implementation proposal so feel free to leave it as is and we can take it forward from here.

File: app/src/main/AndroidManifest.xml
Comment (xenom34) [reply app/src/main/AndroidManifest.xml:15]: Indeed, I did not see that the permission was already there 😅

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (xenom34) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt]: I moved the code in a seperate function, as suggested. The function will be executed after setContentView

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (xenom34) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt]: Changed to 5000ms

File: app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt
Comment (xenom34) [reply app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt]: Deleted

**Comment by xenom34** [comment]
"Hi @ashiagr ! I tried to modify the code based on your feedback. Hoping it meets your expectations and that the CI is now satisfied."

**Comment by ashiagr** [comment]
Thank you so much for the changes, @xenom34! We're moving closer to merging it 🙌

Just a couple of more observations, I noticed that while the app asks for permission on a fresh install, it doesn't handle the [permission denial](https://developer.android.com/training/permissions/requesting#handle-denial) very well: 

1. If I deny the notification permission, it keeps displaying the snackbar every time I restart the app.

https://user-images.githubusercontent.com/1405144/217230616-c2ecbfdc-2435-4a2f-8547-555c86c83012.mp4

2. Reading [this comment](https://developer.android.com/training/permissions/requesting#kotlin), we shouldn’t be linking to `Settings` (from the snackbar) to convince the user to change their decision

```
// [...] respect the user's decision. Don't link to system
// settings in an effort to convince the user to change their
// decision.
```

To fix 2., how about we show an alert with a specific message about why we need notification permission? We can get someone to proofread the message.
alert

To fix 1., we can use "Retry" button on the alert to show the permission dialog again and if the user decides to disable it again, let the system take it as an [implicit "don't ask again"](https://developer.android.com/training/permissions/requesting#handle-denial), something like in this video:

https://user-images.githubusercontent.com/1405144/217231830-c0db54db-3109-42ec-a225-238dde4ed5ea.mp4

Here's a patch I tried on your branch: [notifications_permission.patch](https://github.com/Automattic/pocket-casts-android/files/10674467/notifications_permission.patch)

Also, let me ping @mchowning for another opinion 🙂

**Comment by ashiagr** [comment]
In order to unblock this PR, I just added a check to show the snackbar only once when a user has denied permission.
Also, I continued displaying the snackbar with an action to go to `Settings` as it needs user interaction and doesn't forcibly go to `Settings`.

I'll wait for any comments till tomorrow and merge the PR.

**Review by ashiagr — APPROVED** [review]

**Pull request merged**
